### PR TITLE
(port)  Vampirism virology symptom a new traitor only virology symptom

### PR DESCRIPTION
**Pull request by boodaliboo**
original pr:https://github.com/BeeStation/BeeStation-Hornet/pull/5624/files
You can use blood on the floor to heal bloodloss, and drink or use any bloodtype. transmission threshold to heal brute. your blood is slowly consumed. has an undocumented threshold that causes you to consume blood from nearby corpses to gain a defrib-analagous revive.

in the uplink for 6tc
:cl:  
rscadd: Added new virus
/:cl:

### CODE REVIEW

**Comment by Lichton** [comment]
Love the concept, great work here! Hope to see it soon

**Comment by boodaliboo** [comment]
can anyone comment on why it aint compiling

**Comment by VaelophisNyx** [comment]
ok so realtalk, this is problematic for a few reasons

1. Blood is incredibly common. if you can get it from the floor alone, that might be too strong compared to being required to drink it.
2. The automated revival part is INSANE. Like, flat out, way too strong. Not sure that can be kept at all.

alternate PR name suggestion: [Hematophagy](https://en.wikipedia.org/wiki/Hematophagy)

File: code/datums/diseases/advance/symptoms/heal.dm [hunk 1]
```diff
@@ -492,3 +492,240 @@
 		if(L.heal_damage(heal_amt/parts.len, heal_amt/parts.len, null, BODYPART_ORGANIC))
 			M.update_damage_overlays()
 	return 1
+	
+
+/datum/symptom/vampirism
+	name = "Hemetophagy"
```
Comment (VaelophisNyx) [inline code/datums/diseases/advance/symptoms/heal.dm]: Hemetophagy -> Hematophagy

File: code/datums/diseases/advance/symptoms/heal.dm [hunk 2]
```diff
@@ -492,3 +492,240 @@
...
+	var/vampire = FALSE
+	var/mob/living/carbon/human/bloodbag
+		threshold_descs = list(
+		"Transmission 4" = "Host appears to die when falling into a coma.",
+		"Transmission 6" = "The virus aggressively assimilates blood, resulting in contiguous blood pools being absorbed by the virus, as well as sucking blood out of open wounds of subjects in physical contact with the host.",
+		"Stage Speed 7" = "The virus grows more aggressive, assimilating blood and healing at a faster rate, but also draining the host's blood quicker",
+	)
+	
+	
+	threshold_descs = list("Transmission 4" "The virus recycles excess absorbed blood into restorative biomass, healing brute damage."
```
Comment (VaelophisNyx) [inline code/datums/diseases/advance/symptoms/heal.dm]: put the stuff after the ( on a newline

File: code/datums/diseases/advance/symptoms/heal.dm [hunk 3]
```diff
@@ -492,3 +492,240 @@
...
+		H.dna.blood_type = "U"
+
+/datum/symptom/vampirism/Activate(datum/disease/advance/A)
+	if(!..())
+		return
+	var/mob/living/carbon/M = A.affected_mob
+	switch(A.stage)
+		if(1 to 4)
+			if(prob(5))
+				to_chat(M, "<span class='warning'>[pick("You feel cold...", "You feel a bit thirsty", "It dawns upon you that every single human on this station has warm blood pulsing through their veins.")]</span>")
```
Comment (VaelophisNyx) [inline code/datums/diseases/advance/symptoms/heal.dm]: "You feel a craving for iron?", "you could use a sip of...blood?", "the thought of biting a neck crosses your mind..."

File: code/datums/diseases/advance/symptoms/heal.dm [hunk 4]
```diff
@@ -492,3 +492,240 @@
...
+					else if(bruteheal && M.getBruteLoss())
+						bloodpoints -= 1
+						M.heal_overall_damage(2, required_status = BODYTYPE_ORGANIC)
+					if(prob(60) && !M.stat) 
+						bloodpoints -- //you cant just accumulate blood and keep it as a battery of healing. the quicker the symptom is, the faster your bloodpoints decay
+				else if(prob(20) && M.blood_volume >= BLOOD_VOLUME_BAD)//the virus continues to extract blood if you dont have any stored up. higher probability due to BP value
+					M.blood_volume = (M.blood_volume - 1)
+
+			if(!bloodpoints && prob(3))
+				to_chat(M, "<span class='warning'>[pick("You feel a pang of thirst.", "No food can sate your hunger", "Blood...")]</span>")
```
Comment (VaelophisNyx) [inline code/datums/diseases/advance/symptoms/heal.dm]: "You need blood!", "Your throat is dry...", "Your heart flutters alarmingly..."

**Comment by VaelophisNyx** [comment]
Problem 3

**why are we giving random shitters the free ability to suck other peoples' blood.**
Vampire was removed from Pride's Mirror just recently for this very problem
even if this is the automated one for revive, this is abusable to be a griefing little shit

**Comment by ghost** [comment]
It's morbin' time

**Comment by boodaliboo** [comment]
> Problem 3
> 
> **why are we giving random shitters the free ability to suck other peoples' blood.** Vampire was removed from Pride's Mirror just recently for this very problem even if this is the automated one for revive, this is abusable to be a griefing little shit

we already give the ability to mass kill and poison people with nanites eye pop with viro and you can make mass armys loyal to you with xenobio

**Comment by VaelophisNyx** [comment]
yes however none of those are flat out antag abilities like bloodsucking

**Comment by boodaliboo** [comment]
> yes however none of those are flat out antag abilities like bloodsucking
what

**Comment by CMOisLing** [comment]
Maybe add it as a viro traitor specific Item. It could be used to buff you specifically or just create chaos for sec to deal with. It would be nice way to have a traitor viro that doesnt "accidental" kill off half the station with Covid 9000.

**Comment by boodaliboo** [comment]
> Maybe add it as a viro traitor specific Item. It could be used to buff you specifically or just create chaos for sec to deal with. It would be nice way to have a traitor viro that doesnt "accidental" kill off half the station with Covid 9000.

thats a good idea ill go about doing that as my other traitor viro idea clearly wont end up being merged

File: code/datums/diseases/advance/symptoms/heal.dm (hunk 3 above)
Comment (boodaliboo) [reply code/datums/diseases/advance/symptoms/heal.dm]: ty

File: code/datums/diseases/advance/symptoms/heal.dm [hunk 5]
```diff
@@ -492,3 +492,235 @@
...
+					else if(bruteheal && M.getBruteLoss())
+						bloodpoints -= 1
+						M.heal_overall_damage(2, required_status = BODYTYPE_ORGANIC)
+					if(prob(60) && !M.stat) 
+						bloodpoints -- //you cant just accumulate blood and keep it as a battery of healing. the quicker the symptom is, the faster your bloodpoints decay
+				else if(prob(20) && M.blood_volume >= BLOOD_VOLUME_BAD)//the virus continues to extract blood if you dont have any stored up. higher probability due to BP value
+					M.blood_volume = (M.blood_volume - 1)
+
+			if(!bloodpoints && prob(3))
+				to_chat(M, "<span class='warning'>[pick("You need blood!".", "Your throat is dry...", "Your heart flutters alarmingly...")]</span>")
```
Comment (redmoogle) [inline code/datums/diseases/advance/symptoms/heal.dm:583]:
```suggestion
				to_chat(M, span_warning("[pick("You need blood!", "Your throat is dry...", "Your heart flutters alarmingly...")]"))
```

**Review by redmoogle — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Why

**Comment by boodaliboo** [comment]
> Why

i am constantly insulted in the public chat when i dont even insult them back but skrem and theos enter chat litterally any time im join and then constantly talk about me when im not alongside multiple people who just follow along fuck you theos your a bitch who made this into personal insults for no reason because i made one comment against the people who constantly insult me

**Comment by boodaliboo** [comment]
anyone can finish this all thats needed is redmoogles commit to fix it

**Pull request closed without merging**